index.d.ts

```diff
@@ -0,0 +1,4 @@
+export declare function verifyPaddleWebhook(
+    publicKey: string,
+    webhookData: Readonly<Record<string, string | string[] | undefined>>,
```

Comment:
I am no TS expert but not sure that the `string[] | undefined` part is necessary for this.

What do you think to the following?:

```suggestion
    webhookData: Readonly<Record<string, string>>,
```

Reply:
I first tried using a `Record<string, string>` to match your docstring, but the return type of `querystring.parse` (which is `Record<string, string | string[] | undefined>`) cannot be assigned to that type. Then I tried widening the function type to `Record<string, string | string[]>` but still no dice, because of the TS compiler setting `--strictNullChecks`.

When I widened the function type to match the return type of `querystring.parse` exactly, I was able to compile

Reply:
Ah yep I see.

This also makes me realise that `querystring.parse` happens to convert numbers to strings, whereas other approaches may not necessarily do that. This is why the implementation at [line25](https://github.com/daveagill/verify-paddle-webhook/blob/master/index.js#L25) is allowing for numbers (and null).

All it really needs from the value is toString().

Given the above, perhaps one of these makes sense?:

```suggestion
    webhookData: Readonly<Record<string, any>>,
```

```suggestion
    webhookData: Readonly<object>,
```

Reply:
Hmm, I see your point! 

> All it really needs from the value is toString()

We can work with this information, this should represent the type you just described
```suggestion
    webhookData: Readonly<Record<string, { toString(): string } | undefined | null>>,
```

It's an odd looking function, but then again we're gradually typing a dynamic codebase.
I tested this proposal locally against both strings and numbers with the following code:

``` typescript
import { verifyPaddleWebhook } from 'verify-paddle-webhook'
import q from 'querystring'

const a = q.parse('')
const test1 = verifyPaddleWebhook('', a)
const test2 = verifyPaddleWebhook('', { 'hi': 5 })
```

compiles like a charm!

Reply:
Cool! Let's go with that then 👍

Reply:
Hey @daveagill would you mind merging this in?

Reply:
Hey @amchelle - Yep happy to but (correct me if I'm wrong) I believe you need to accept the above code suggestion into your PR first, I don't think I can do it as I don't have write access to your branch/repo.